pyomo/core/kernel/component_interface.py

```diff
@@ -210,6 +212,8 @@ def __str__(self):
             for a way to generate a static set of component
             names.
         """
+        if ICategorizedObject.labeler:
```

Comment:
So far, I've managed to keep these newer files in Kernel 100% covered with smoke unit tests (no big external dependencies). This change also makes the doc string incorrect. I'm being pedantic here, but I don't want these kind of low-level hacks to start creeping into Kernel without testing and documentation.

An update to the doc string and a unit test for the variable would probably do the trick (pyomo/core/tests/unit/test_component_variable.py). Perhaps @jsirrola can chime in here.